### DISCUSSION TITLE
Sharing Sinks Across Multiple Configuration Files

### DISCUSSION THREAD

**Discussion by arch-xtof** [vectordotdev/vector#21304] [Q&A] [answered]
Hello!

I currently work with multiple configuration files in Vector, each handling separate pipelines. It’s been great for modularizing the setup, but I’ve encountered a situation where I’m collecting metrics from each pipeline and need to expose them on the same port, meaning I need to share the same sink.

I referenced sources and transforms from other configuration files to achieve this, and it works, but I’d like to know if this is considered acceptable practice. Are there any potential pitfalls or issues I should be aware of when sharing sinks across multiple config files?

I appreciate any suggestions!

**Comment by jszwedko** [comment, marked as answer]
When Vector loads configuration files it essentially just concatenates them all together so there is no issues I can think of that would result from sharing a sink between configuration files.

**Reply by arch-xtof** [reply]
Thank you! I did some experiments and it indeed works without issues.
